stan/math/prim/prob/discrete_range_rng.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef STAN_MATH_PRIM_PROB_DISCRETE_RANGE_RNG_HPP
+#define STAN_MATH_PRIM_PROB_DISCRETE_RANGE_RNG_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/max_size.hpp>
+#include <boost/random/uniform_int_distribution.hpp>
+#include <boost/random/variate_generator.hpp>
+
+namespace stan {
+namespace math {
+
+/** \ingroup prob_dists
+ * Return a discrete random variate for the given lower and upper bounds
+ * (inclusive) using the specified random number generator.
+ *
+ * `lower` and `upper` can each be a scalar or a one-dimensional container.
+ * Any non-scalar inputs must be the same size.
+ *
+ * @tparam T_lower type of lower bound
+ * @tparam T_upper type of upper bound
+ * @tparam RNG type of random number generator
+ *
+ * @param lower lower bound
+ * @param upper upper bound
+ * @param rng random number generator
+ * @return A (sequence of) discrete random variate(s) between `lower` and
```

Comment:
The return needs to be a single integer when both arguments are integers, and a standard vector of integers otherwise.  Is that what this is supposed to mean?  Is it how the other RNGs are documnted?  I find it a bit confusing, but if it's like all the other ones, we should go with consistency for now.